src/common/snippets/include/snippets/lowered/loop_manager.hpp

```
@@ -154,56 +214,139 @@ class LinearIR::LoopManager {
-
-    // When the previous expression was replaced with new expressions (decomposition), the method updates the corresponding Loop.
-    // If ports of decomposed expression were the Loop ports, these Loop ports may be updated by parameters `entries` and `exits`
-    // Note: This method should be removed when Softmax decomposition will be moved on data flow pipeline since
-    //       all decompositions should be call on this pipeline
+    /**
+     * @brief When the previous expression was replaced with new expressions (decomposition), the method updates the corresponding Loop.
+     *        If ports of decomposed expression were the Loop ports, these Loop ports may be updated by parameters `entries` and `exits`
+     *        Note: This method should be removed when Softmax decomposition will be moved on data flow pipeline since
+     *              all decompositions should be call on this pipeline
+     * @param new_expr_begin the first expression iterator
+     * @param new_expr_end the next iterator after the last expression
+     * @param decomposed_expr the expression that is decomposed into several other exprs
+     * @param loop_id the new expression will be marked by this loop ID
+     * @param new_entries new input loop ports
+     * @param new_exits new output loop ports
+     */
     void expression_replacement(constExprIt new_expr_begin, constExprIt new_expr_end, const ExpressionPtr& decomposed_expr,
                                 size_t loop_id, const std::vector<ExpressionPort>& new_entries, const std::vector<ExpressionPort>& exits);
+    /**
+     * @brief Find iterators of first entry loop point and last exit point (bounds of Loop) when explicit Loop exprs haven't been inserted in LinearIR
+     *        Note: This method cannot be called when explicit loop exprs are inserted since Loop bounds are defined by LoopBegin and LoopEnd after the insertion!
```

When we design a method, we should do our best to guarantee that it yields appropriate and expected result. If we can not guarantee it, we should throw an exception. Unfortunately, it is not enough to write a comment to prevent incorrect usage of a method 🥲 
We need to either generalize it to always work correctly (which is a preferred way) or to throw an exception, when it is used incorrectly

---

LoopBegin/End with a given id can't be inserted at any position, right? (we still nave our entry/exit points). Then do we really need to iterate through the whole IR?
Let's continue this discussion in loop_manager.cpp